packages/gatsby/src/commands/develop-process.ts

```diff
@@ -681,6 +686,9 @@ module.exports = async (program: IProgram): Promise<void> => {
       webpackActivity = null
     }
 
+    await pageDataUtil.flush()
```

Comment:
Considering we basically always flush before `markAsDone`, should it be part of `webpackStatusUtil`? Like a webpack transaction.

Reply:
Could do so but wanted to keep them separate for now since state machine

Reply:
I'd prefer to keep current code. `webpackStatusUtil` is really just glorified boolean (just with extra sugar on top to have bit more semantic meaning), moving more logic there just spread that logic between multiple files unnecessarily (IMO). And we really only call `markAsDone` in 1 place for develop and in 1 place for build